tests/experimental/wgit/test_sha1_store.py

```diff
@@ -3,148 +3,106 @@
 # This source code is licensed under the BSD license found in the
 # LICENSE file in the root directory of this source tree.
 
-from dataclasses import dataclass
-import json
 import os
 from pathlib import Path
 import shutil
 
 import pytest
+import torch
+from torch import nn
 
-from fairscale.experimental.wgit.repo import Repo
-from fairscale.experimental.wgit.sha1_store import SHA1_store
+from fairscale.experimental.wgit.sha1_store import SHA1_Store
 
+# Get the absolute path of the parent at the beginning before any os.chdir(),
+# so that we can proper clean it up at any CWD.
+PARENT_DIR = Path("sha1_store_testing").resolve()
 
-@pytest.fixture
-def sha1_configs():
-    @dataclass
-    class Sha1StorePaths:
-        test_dirs = Path("temp_wgit_testing/.wgit")
-        test_path = Path.cwd().joinpath(test_dirs)
-        sha1_ref = test_path.joinpath("sha1_refs.json")
-        chkpt1a_dir = test_path.joinpath("checkpoint_1a")
-        chkpt1b_dir = test_path.joinpath("checkpoint_1b")
-        chkpt1c_dir = test_path.joinpath("checkpoint_1c")
-        checkpoint_1a = test_path.joinpath("checkpoint_1a", "checkpoint_1.pt")
-        checkpoint_1b = test_path.joinpath("checkpoint_1b", "checkpoint_1.pt")
-        checkpoint_1c = test_path.joinpath("checkpoint_1c", "checkpoint_1.pt")
-        checkpoint_2 = test_path.joinpath("checkpoint_1a", "checkpoint_2.pt")
-        checkpoint_3 = test_path.joinpath("checkpoint_1a", "checkpoint_3.pt")
-        metadata_1 = test_path.joinpath("checkpoint_1.pt")
-        metadata_2 = test_path.joinpath("checkpoint_2.pt")
-        metadata_3 = test_path.joinpath("checkpoint_3.pt")
 
-    return Sha1StorePaths
+@pytest.fixture(scope="function")
+def sha1_store(request):
+    """A fixture for setup and teardown.
 
+    This only runs once per test function. So don't make this too slow.
 
-@pytest.fixture
-def sha1_store(sha1_configs):
-    repo = Repo(sha1_configs.test_path.parent, init=False)
-    sha1_store = SHA1_store(sha1_configs.test_dirs, init=False)
-    return repo, sha1_store
+    Tests must be written in a way that either all of the tests run
+    in the order they appears in this file or a specific test is
+    run separately by the user. Either way, the test should work.
```

Comment:
this could be something to pay attention to for existing and future tests. @riohib